From d4e727aeda6b2bbde0af262cdf327c9e146e385b Mon Sep 17 00:00:00 2001
From: Dylan Hung <dylan_hung@aspeedtech.com>
Date: Tue, 5 Jul 2022 16:43:21 +0800
Subject: [PATCH 10/30] i3c: slave: mqueue: add build assertion for init
 priority

Add a build assertion to ensure the i3c controller is initialized prior
to the target/slave device driver.

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I32b6a30f48b59735052542fe8e63343fc96e7ba3
---
 drivers/i3c/slave/i3c_slave_mqueue.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/i3c/slave/i3c_slave_mqueue.c b/drivers/i3c/slave/i3c_slave_mqueue.c
index a6908c0b60..ad511d6740 100644
--- a/drivers/i3c/slave/i3c_slave_mqueue.c
+++ b/drivers/i3c/slave/i3c_slave_mqueue.c
@@ -177,6 +177,9 @@ static void i3c_slave_mqueue_init(const struct device *dev)
 	i3c_slave_register(obj->i3c_controller, &slave_data);
 }
 
+BUILD_ASSERT(CONFIG_I3C_SLAVE_INIT_PRIORITY > CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
+	     "I3C controller must be initialized prior to target device initialization");
+
 #define I3C_SLAVE_MQUEUE_INIT(n)                                                                   \
 	static int i3c_slave_mqueue_config_func_##n(const struct device *dev);                     \
 	static const struct i3c_slave_mqueue_config i3c_slave_mqueue_config_##n = {                \
-- 
2.24.1

